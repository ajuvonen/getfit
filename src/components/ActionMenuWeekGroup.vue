<script setup lang="ts">
import {storeToRefs} from 'pinia';
import type {Training} from '@/types';
import useWeekDays from '@/hooks/weekdays';
import {useScheduleStore} from '@/stores/schedule';

const props = defineProps<{
  action: 'move' | 'copy';
  training: Training;
}>();

const scheduleStore = useScheduleStore();
const {settings, weeks} = storeToRefs(scheduleStore);
const {moveTraining, copyTraining} = scheduleStore;

const {getDisplayWeekNumber, getShortDate, weekdays} = useWeekDays();

const menuAction = props.action === 'move' ? moveTraining : copyTraining;
const menuIcon = props.action === 'move' ? 'mdi-arrow-all' : 'mdi-content-copy';
</script>
<template>
  <v-list-group>
    <template v-slot:activator="{props}">
      <v-list-item
        v-bind="props"
        :title="$t(`trainingCard.${action}`)"
        :prepend-icon="menuIcon"
      />
    </template>
    <v-list-group v-for="(week, weekIndex) in weeks" :key="week.id">
      <template v-slot:activator="{props}">
        <v-list-item
          v-bind="props"
          :title="$t('weekCalendar.weekTitle', [getDisplayWeekNumber(weekIndex)])"
        />
      </template>
      <v-list-item
        v-for="(day, dayIndex) in weekdays"
        :disabled="week.id === training.weekId && dayIndex === training.dayIndex"
        :key="day"
        :title="day"
        @click="menuAction(training, week.id, dayIndex)"
      >
        <template v-if="settings.startDate" #append>
          <span class="ml-4">{{ getShortDate(dayIndex, weekIndex) }}</span>
        </template>
      </v-list-item>
    </v-list-group>
  </v-list-group>
</template>
<style lang="scss" scoped></style>
