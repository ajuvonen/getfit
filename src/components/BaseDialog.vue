<script setup lang="ts">
import useScreen from '@/hooks/screen';

withDefaults(
  defineProps<{
    show: boolean;
    title: string;
    useFullScreen?: boolean;
  }>(),
  {
    useFullScreen: true,
  },
);

const {isSmallScreen} = useScreen();
</script>
<template>
  <v-dialog
    :model-value="show"
    :width="!isSmallScreen ? 'auto' : undefined"
    :fullscreen="useFullScreen && isSmallScreen"
    :transition="isSmallScreen ? 'dialog-bottom-transition' : false"
    :min-width="!isSmallScreen ? 500 : undefined"
    persistent
  >
    <v-card>
      <v-card-title>{{ title }}</v-card-title>
      <v-card-text>
        <slot name="content" />
      </v-card-text>
      <v-card-actions><slot name="actions" /></v-card-actions>
    </v-card>
  </v-dialog>
</template>
<style scoped></style>
