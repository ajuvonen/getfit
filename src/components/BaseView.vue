<script setup lang="ts">
import useScreenSize from '@/hooks/screenSize';

defineProps<{
  title: string;
  guide: string;
}>();

const {isSmallScreen, isMediumScreen} = useScreenSize();
</script>
<template>
  <v-card
    color="rgba(255,255,255,0.9)"
    class="mb-4"
    :rounded="isSmallScreen || isMediumScreen ? 0 : 'rounded'"
  >
    <v-card-title class="d-print-none">
      <h1 class="text-h4">{{ title }}</h1>
    </v-card-title>
    <v-card-text>
      <p class="d-flex align-center d-print-none mb-4">
        <v-icon icon="mdi-information-box-outline" />
        <span class="ml-4">{{ guide }}</span>
      </p>
      <slot name="content" />
    </v-card-text>
    <v-card-actions v-if="$slots.actions">
      <slot name="actions" />
    </v-card-actions>
  </v-card>
</template>