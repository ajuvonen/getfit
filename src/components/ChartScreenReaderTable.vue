<script lang="ts" setup>
defineProps<{
  id: string,
  title: string,
  labels: string[],
  data: number[],
}>();
</script>
<template>
  <table class="d-sr-only" :id="id">
    <caption>{{ title }}</caption>
    <thead>
      <tr>
        <th v-for="label in labels" :key="label">{{ label }}</th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td v-for="count in data" :key="count">{{ count }}</td>
      </tr>
    </tbody>
  </table>
</template>
