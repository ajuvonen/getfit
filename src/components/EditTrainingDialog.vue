<script setup lang="ts">
import {computed} from 'vue';
import {storeToRefs} from 'pinia';
import {useI18n} from 'vue-i18n';
import {useVuelidate} from '@vuelidate/core';
import {required, between, maxLength, integer, helpers} from '@vuelidate/validators';
import type {VForm} from 'vuetify/components';
import {Intensity} from '@/types';
import {useScheduleStore} from '@/stores/schedule';
import {useAppStateStore} from '@/stores/appState';
import {ACTIVITIES} from '@/constants';
import {getIntensityColor, getIcon, getValidationErrors, decimalRegex} from '@/utils';
import useScreenSize from '@/hooks/screenSize';
import useLocalizedActivities from '@/hooks/localizedActivities';

const scheduleStore = useScheduleStore();
const {settings} = storeToRefs(scheduleStore);
const {addOrEditTraining} = scheduleStore;

const appStateStore = useAppStateStore();
const {trainingDialogOpen, trainingData} = storeToRefs(appStateStore);

const {t} = useI18n();

const {isSmallScreen} = useScreenSize();

const tickLabels = computed(() => ({
  [Intensity.LIGHT]: t('intensities.LIGHT'),
  [Intensity.NORMAL]: t('intensities.NORMAL'),
  [Intensity.MEDIUM]: t('intensities.MEDIUM'),
  [Intensity.HEAVY]: t('intensities.HEAVY'),
}));

const {localizedAvailableActivities} = useLocalizedActivities();

const handleSave = async () => {
  if (await v$.value.$validate()) {
    addOrEditTraining(trainingData.value);
    resetAndClose();
  }
};

const rules = computed(() => ({
  title: {maxLength: maxLength(30)},
  location: {maxLength: maxLength(40)},
  description: {maxLength: maxLength(2000)},
  duration: {
    required,
    between: between(0, settings.value.unitOfTime === 'm' ? 360 : 6),
    precision:
      settings.value.unitOfTime === 'm'
        ? integer
        : helpers.withMessage(t('errors.invalidPrecision'), decimalRegex),
  },
  activity: {required},
}));

const v$ = useVuelidate(rules, trainingData);

const resetAndClose = () => {
  v$.value.$reset();
  trainingDialogOpen.value = false;
};
</script>
<template>
  <v-dialog
    v-model="trainingDialogOpen"
    :width="!isSmallScreen ? 'auto' : undefined"
    :fullscreen="isSmallScreen"
    :transition="isSmallScreen ? 'dialog-bottom-transition' : false"
    :min-width="!isSmallScreen ? 500: undefined"
    persistent
  >
    <v-card>
      <v-card-title>{{ $t('editTraining.title') }}</v-card-title>
      <v-card-text>
        <v-form>
          <v-label for="edit-training-activity">{{ $t('editTraining.activity') }}</v-label>
          <v-select
            id="edit-training-activity"
            v-model="trainingData.activity"
            :error-messages="getValidationErrors(v$.activity)"
            :items="localizedAvailableActivities"
            eager
            data-test-id="edit-training-activity"
            @input="v$.activity.$touch"
            @blur="v$.activity.$touch"
          >
            <template #selection="{item: {title, value}}">
              <div v-if="value">
                <v-icon :icon="getIcon(ACTIVITIES, value)" />
                {{ title }}
              </div>
            </template>
            <template #item="{item: {title, value}, props}">
              <v-list-item v-bind="props" :data-test-id="`edit-training-activity-${value}`">
                <template #title>
                  <v-icon :icon="getIcon(ACTIVITIES, value)" />
                  {{ title }}
                </template>
              </v-list-item>
            </template>
          </v-select>
          <v-text-field
            v-model="trainingData.title"
            :label="t('editTraining.trainingTitle')"
            :error-messages="getValidationErrors(v$.title)"
            maxlength="30"
            counter
            data-test-id="edit-training-title"
            @input="v$.title.$touch"
            @blur="v$.title.$touch"
          />
          <v-text-field
            v-model="trainingData.location"
            :label="t('editTraining.location')"
            :error-messages="getValidationErrors(v$.location)"
            maxlength="40"
            counter
            data-test-id="edit-training-location"
            @input="v$.location.$touch"
            @blur="v$.location.$touch"
          />
          <v-text-field
            v-model.number="trainingData.duration"
            :error-messages="getValidationErrors(v$.duration)"
            :suffix="settings.unitOfTime"
            :label="t('editTraining.duration')"
            class="edit-training-duration"
            type="number"
            data-test-id="edit-training-duration"
            @input="v$.duration.$touch"
            @blur="v$.duration.$touch"
          />
          <v-textarea
            v-model="trainingData.description"
            :label="t('editTraining.description')"
            :error-messages="getValidationErrors(v$.description)"
            clearable
            maxlength="2000"
            no-resize
            counter
            data-test-id="edit-training-description"
            @input="v$.description.$touch"
            @blur="v$.description.$touch"
          />
          <v-label for="edit-training-intensity">{{ $t('editTraining.intensity') }}</v-label>
          <v-slider
            id="edit-training-intensity"
            v-model="trainingData.intensity"
            :ticks="tickLabels"
            :color="getIntensityColor(trainingData.intensity)"
            max="3"
            step="1"
            show-ticks="always"
            tick-size="4"
            data-test-id="edit-training-intensity"
          ></v-slider>
        </v-form>
      </v-card-text>
      <v-card-actions>
        <v-btn
          prepend-icon="mdi-content-save"
          variant="outlined"
          color="primary"
          data-test-id="edit-training-save-button"
          @click="handleSave()"
          >{{ $t('editTraining.save') }}</v-btn
        >
        <v-btn
          prepend-icon="mdi-close"
          data-test-id="edit-training-close-button"
          variant="tonal"
          @click="resetAndClose"
          >{{ $t('editTraining.closeDialog') }}</v-btn
        >
      </v-card-actions>
    </v-card>
  </v-dialog>
</template>

<style lang="scss" scoped>
:deep(.edit-training-duration .v-input__control) {
  max-width: 200px;
  input[type='number'] {
    appearance: textField;
  }
}
</style>
