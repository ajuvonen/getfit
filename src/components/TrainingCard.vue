<script setup lang="ts">
import {storeToRefs} from 'pinia';
import {useScheduleStore} from '@/stores/schedule';
import {Intensity, type Training} from '@/types';
import {getIcon, getIntensityColor} from '@/utils';
import {ACTIVITIES} from '@/constants';
import {useI18n} from 'vue-i18n';
import TrainingCardSummary from '@/components/TrainingCardSummary.vue';
import TrainingCardActions from '@/components/TrainingCardActions.vue';
import {useAppStateStore} from '@/stores/appState';

defineProps<{
  training: Training;
}>();

const scheduleStore = useScheduleStore();
const {schedule} = storeToRefs(scheduleStore);
const appStateStore = useAppStateStore();
const {isSummaryShown} = storeToRefs(appStateStore);
const {toggleSummaryShown} = appStateStore;
const {t} = useI18n();
</script>
<template>
  <v-card class="training-card mx-2 mt-4 mb-2" min-width="400" max-width="400">
    <v-card-item
      class="training-card__title-wrapper text-white"
      :style="{'background-color': getIntensityColor(training.intensity)}"
    >
      <v-icon
        :icon="getIcon(ACTIVITIES, training.activity)"
        :title="t(`activities.${training.activity}`)"
        :aria-label="t(`activities.${training.activity}`)"
        :color="getIntensityColor(training.intensity)"
        class="training-card__activity-icon"
        size="x-large"
      />
      <v-card-title class="d-flex ml-15 flex-column justify-top">
        <div class="training-card__title">
          {{ training.title || t(`activities.${training.activity}`) }}
        </div>
        <div v-if="training.duration" class="training-card__duration text-subtitle-2">
          <v-icon icon="mdi-timer" :aria-label="t('trainingCard.duration')" />
          {{ training.duration }} {{ schedule.unitOfTime }}
        </div>
        <div v-if="training.location" class="training-card__location text-subtitle-2">
          <v-icon icon="mdi-map-marker" :aria-label="t('trainingCard.location')" />
          {{ training.location }}
        </div>
        <div class="training-card__intensity text-subtitle-2">
          <v-icon icon="mdi-speedometer" :aria-label="t('trainingCard.intensity')" />
          {{ t(`intensities.${Intensity[training.intensity]}`) }}
        </div>
      </v-card-title>
    </v-card-item>
    <v-card-text class="training-card__text">
      <training-card-summary :training="training" />
    </v-card-text>
    <v-card-actions class="justify-center">
      <training-card-actions v-if="!schedule.lockSchedule" :training="training" />
      <v-btn
        v-else
        class="training-card__show-summary-button"
        @click="toggleSummaryShown(training.id)"
        >{{
          t(isSummaryShown(training.id) ? 'trainingCard.hideSummary' : 'trainingCard.showSummary')
        }}</v-btn
      >
    </v-card-actions>
  </v-card>
</template>
<style lang="scss" scoped>
.v-card-item {
  cursor: move;
}

.training-card__title-wrapper {
  position: absolute;
  left: 0;
  right: 0;
  top: 0;
  height: 7rem;
}

.training-card__subtitle {
  opacity: 1;
}

.training-card__text {
  margin-top: 7.5rem;
}

.training-card__activity-icon {
  background: white;
  border-radius: 50%;
  border-top-left-radius: 0;
  border-bottom-left-radius: 0;
  position: absolute;
  left: 0;
  top: 0;
  height: 100%;
  padding: 0rem 2rem 0 3.5rem;
  margin-left: -1.5rem;
}
</style>
