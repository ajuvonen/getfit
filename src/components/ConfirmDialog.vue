<script setup lang="ts">
import {storeToRefs} from 'pinia';
import {useAppStateStore} from '@/stores/appState';
import BaseDialog from '@/components/BaseDialog.vue';

const {confirmDialogOpen, confirmText, confirmAction} = storeToRefs(useAppStateStore());
</script>
<template>
  <BaseDialog
    :open="confirmDialogOpen"
    :use-full-screen="false"
    :title="$t('general.confirmTitle')"
  >
    <template #content>{{ confirmText }}</template>
    <template #actions>
      <v-btn
        color="primary"
        variant="outlined"
        prepend-icon="mdi-check"
        data-test-id="confirm-dialog-confirm-button"
        @click="confirmAction(), (confirmDialogOpen = false)"
        >{{ $t('general.confirm') }}</v-btn
      >
      <v-btn
        prepend-icon="mdi-close"
        data-test-id="confirm-dialog-cancel-button"
        @click="confirmDialogOpen = false"
        >{{ $t('general.cancel') }}</v-btn
      >
    </template>
  </BaseDialog>
</template>
