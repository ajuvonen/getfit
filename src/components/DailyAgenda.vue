<script lang="ts" setup>
import {computed} from 'vue';
import {storeToRefs} from 'pinia';
import {DateTime} from 'luxon';
import {useScheduleStore} from '@/stores/schedule';
import TrainingCard from '@/components/TrainingCard.vue';

const scheduleStore = useScheduleStore();
const {weeks, settings, getTrainingsCount, getCompletedTrainingsCount} = storeToRefs(scheduleStore);

const dailyTrainings = computed(() => {
  const startDate = DateTime.fromJSDate(settings.value.startDate!);
  const difference = Math.ceil(startDate.diffNow('weeks').weeks);
  if (difference < 1) {
    const targetWeek = weeks.value[Math.abs(difference)];
    if (targetWeek) {
      const targetDay = (DateTime.now().weekday - (settings.value.startsOnSunday ? 0 : 1)) % 7;
      return targetWeek.trainings.filter(({dayIndex}) => dayIndex === targetDay);
    }
  }
  return [];
});
</script>
<template>
  <v-progress-circular
    :model-value="(getCompletedTrainingsCount / getTrainingsCount) * 100"
    class="mx-auto"
    size="150"
    width="15"
  >
    <v-fab-transition>
      <div v-if="getTrainingsCount > getCompletedTrainingsCount" class="d-flex flex-column">
        <span class="text-h4 daily-agenda__progress">{{ getCompletedTrainingsCount }} / {{ getTrainingsCount }}</span>
        <span class="text-caption text-uppercase">{{ $t('home.progress') }}</span>
      </div>
      <v-icon v-else class="daily-agenda__completed-icon" icon="$trophyOutline" size="70" />
    </v-fab-transition>
  </v-progress-circular>
  <h1 class="mt-10 text-h3 text-center">
    {{ $t('home.dailyAgenda') }}
  </h1>
  <div class="daily-agenda has-scroll mt-10 d-flex">
    <TrainingCard
      v-for="training in dailyTrainings"
      :key="training.id"
      :training="training"
      simple
    />
    <p v-if="!dailyTrainings.length" class="mx-auto text-h6">{{ $t('home.noTrainings') }}</p>
  </div>
</template>
<style lang="scss" scoped>
.daily-agenda {
  gap: 1rem;
  overflow-x: scroll;
}

.training-card__container:first-child {
  margin-left: auto;
}

.training-card__container:last-child {
  margin-right: auto;
}
</style>
