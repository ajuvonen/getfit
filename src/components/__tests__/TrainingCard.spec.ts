import {mount} from '@vue/test-utils';
import {describe, it, expect, beforeEach} from 'vitest';
import {v4 as uuidv4} from 'uuid';
import {VList} from 'vuetify/components';
import {useScheduleStore} from '@/stores/schedule';
import {useAppStateStore} from '@/stores/appState';
import {Intensity, type Training} from '@/types';
import TrainingCard from '@/components/TrainingCard.vue';
import { getEmptyTraining } from '@/utils';

const basicTraining: Training = getEmptyTraining({
  activity: 'boxing',
  description: 'Sparring at the gym',
  location: 'Total wreck gym',
});

describe('TrainingCard', () => {
  let scheduleStore: ReturnType<typeof useScheduleStore>;
  let appStateStore: ReturnType<typeof useAppStateStore>;

  beforeEach(() => {
    scheduleStore = useScheduleStore();
    appStateStore = useAppStateStore();
  });

  it('mounts', () => {
    const wrapper = mount(TrainingCard, {
      props: {
        training: basicTraining,
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('shows basic data', () => {
    const wrapper = mount(TrainingCard, {
      props: {
        training: basicTraining,
      },
    });

    expect(wrapper.find('.training-card__activity-icon').attributes('aria-label')).toBe('Boxing');
    expect(wrapper.find('.training-card__title').text()).toBe('Boxing');
    expect(wrapper.find('.training-card__duration').text()).toBe('1 h');
    expect(wrapper.find('.training-card__location').text()).toBe('Total wreck gym');
    expect(wrapper.find('.training-card__intensity').text()).toBe('Normal');
  });

  it('shows custom data', async () => {
    scheduleStore.settings.unitOfTime = 'm';
    const wrapper = mount(TrainingCard, {
      props: {
        training: {
          ...basicTraining,
          intensity: Intensity.HEAVY,
          duration: 0,
          location: '',
          title: 'Free fight',
        },
      },
    });

    expect(wrapper.find('.training-card__intensity').text()).toBe('Heavy');
    expect(wrapper.find('.training-card__title').text()).toBe('Free fight');
    expect(wrapper.find('.training-card__duration').text()).toBe('- m');
    expect(wrapper.find('.training-card__location').text()).toBe('-');
});

  it('actions work', async () => {
    const wrapper = mount(TrainingCard, {
      props: {
        training: basicTraining,
      },
    });
    await wrapper.find('.training-card__action-button').trigger('click');
    const actions = wrapper.findComponent(VList);
    await actions.find('.training-card__delete-button').trigger('click');
    await actions.find('.training-card__edit-button').trigger('click');
    expect(scheduleStore.deleteTraining).toHaveBeenCalledOnce();
    expect(appStateStore.openEditTrainingDialog).toHaveBeenCalledOnce();
  });
});
