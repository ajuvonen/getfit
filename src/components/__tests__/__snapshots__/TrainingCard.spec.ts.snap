// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TrainingCard > mounts 1`] = `
"<div data-v-24c3dbf4="" class="training-card__container" style="background: url(/getfit/boxing.jpg);">
  <div data-v-24c3dbf4="" class="v-card v-theme--light v-card--density-default v-card--variant-elevated training-card" style="background-color: rgba(255, 217, 140, 0.8); width: 350px; color: rgb(33, 33, 33);">
    <!---->
    <div class="v-card__loader">
      <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
        <!---->
        <div class="v-progress-linear__background" style="width: 100%;"></div>
        <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
          <div class="v-progress-linear__indeterminate">
            <div class="v-progress-linear__indeterminate long"></div>
            <div class="v-progress-linear__indeterminate short"></div>
          </div>
        </transition-stub>
        <!---->
      </div>
    </div>
    <!---->
    <!---->
    <div data-v-24c3dbf4="" class="v-card-title ml-10 training-card__title"><i data-v-24c3dbf4="" class="mdi-drag-vertical-variant mdi v-icon notranslate v-theme--light" aria-hidden="true"></i><span data-v-24c3dbf4="" class="text-truncate">Boxing</span></div>
    <div data-v-24c3dbf4="" class="v-card-text ml-10">
      <div data-v-24c3dbf4="" class="training-card__icons d-flex align-center px-2"><i data-v-24c3dbf4="" class="mdi-boxing-glove mdi v-icon notranslate v-theme--light v-icon--size-x-large training-card__activity-icon" aria-hidden="true" title="Boxing" aria-label="Boxing"></i></div>
      <div data-v-24c3dbf4="" class="training-card__duration text-subtitle-2"><i data-v-24c3dbf4="" class="mdi-timer mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true" aria-label="Training Duration"></i><span data-v-24c3dbf4="">1 h</span></div>
      <div data-v-24c3dbf4="" class="training-card__intensity text-subtitle-2"><i data-v-24c3dbf4="" class="mdi-speedometer mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true" aria-label="Training Intensity"></i><span data-v-24c3dbf4="">Normal</span></div>
      <div data-v-24c3dbf4="" class="training-card__location text-subtitle-2"><i data-v-24c3dbf4="" class="mdi-map-marker mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true" aria-label="Training Location"></i><span data-v-24c3dbf4="" class="text-truncate">Total wreck gym</span></div>
      <div data-v-cb700901="" data-v-24c3dbf4="" class="v-input v-input--horizontal v-input--center-affix v-input--density-default v-locale--is-ltr v-input--disabled v-radio-group training-card__rating--hidden training-card__rating">
        <!---->
        <div class="v-input__control">
          <!---->
          <div class="v-selection-control-group v-selection-control-group--inline" role="radiogroup" aria-describedby="radio-group-0-messages" aria-label="Training Rating">
            <div data-v-cb700901="" class="v-selection-control v-selection-control--disabled v-selection-control--inline v-selection-control--density-default v-radio">
              <div class="v-selection-control__wrapper">
                <!---->
                <div class="v-selection-control__input"><i class="mdi-star-outline mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i><input disabled="" id="input-3" aria-disabled="true" type="radio" name="radio-group-0" aria-label="1/5" elevation="0" value="1"></div>
              </div>
              <!---->
            </div>
            <div data-v-cb700901="" class="v-selection-control v-selection-control--disabled v-selection-control--inline v-selection-control--density-default v-radio">
              <div class="v-selection-control__wrapper">
                <!---->
                <div class="v-selection-control__input"><i class="mdi-star-outline mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i><input disabled="" id="input-4" aria-disabled="true" type="radio" name="radio-group-0" aria-label="2/5" elevation="0" value="2"></div>
              </div>
              <!---->
            </div>
            <div data-v-cb700901="" class="v-selection-control v-selection-control--disabled v-selection-control--inline v-selection-control--density-default v-radio">
              <div class="v-selection-control__wrapper">
                <!---->
                <div class="v-selection-control__input"><i class="mdi-star-outline mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i><input disabled="" id="input-5" aria-disabled="true" type="radio" name="radio-group-0" aria-label="3/5" elevation="0" value="3"></div>
              </div>
              <!---->
            </div>
            <div data-v-cb700901="" class="v-selection-control v-selection-control--disabled v-selection-control--inline v-selection-control--density-default v-radio">
              <div class="v-selection-control__wrapper">
                <!---->
                <div class="v-selection-control__input"><i class="mdi-star-outline mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i><input disabled="" id="input-6" aria-disabled="true" type="radio" name="radio-group-0" aria-label="4/5" elevation="0" value="4"></div>
              </div>
              <!---->
            </div>
            <div data-v-cb700901="" class="v-selection-control v-selection-control--disabled v-selection-control--inline v-selection-control--density-default v-radio">
              <div class="v-selection-control__wrapper">
                <!---->
                <div class="v-selection-control__input"><i class="mdi-star-outline mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i><input disabled="" id="input-7" aria-disabled="true" type="radio" name="radio-group-0" aria-label="5/5" elevation="0" value="5"></div>
              </div>
              <!---->
            </div>
          </div>
        </div>
        <!---->
        <!---->
      </div>
    </div>
    <div data-v-24c3dbf4="" class="v-card-actions justify-end"><button type="button" class="v-btn v-btn--slim v-theme--light bg-transparent v-btn--density-default v-btn--size-default v-btn--variant-flat training-card__more-button"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span><span class="v-btn__prepend"><i class="mdi-information mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i></span><span class="v-btn__content" data-no-activator="">More</span>
        <!---->
        <!---->
      </button><button type="button" class="v-btn v-btn--slim v-theme--light bg-transparent v-btn--density-default v-btn--size-default v-btn--variant-flat training-card__action-button" aria-label="Actions for training " targetref="[object Object]" aria-haspopup="menu" aria-expanded="false" aria-owns="v-menu-9"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span><span class="v-btn__prepend"><i class="mdi-menu mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i></span><span class="v-btn__content" data-no-activator="">Actions</span>
        <!---->
        <!---->
      </button>
      <!---->
    </div>
    <transition-stub data-v-dc9f9ee2="" data-v-24c3dbf4="" name="expand-transition" appear="false" persisted="false" css="true">
      <!--v-if-->
    </transition-stub>
    <!---->
    <!----><span class="v-card__underlay"></span>
  </div>
</div>"
`;
